src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```
@@ -0,0 +1,35 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> list) {
+        List<FruitTransaction> result = new ArrayList<>(list.size() - 1);
+        for (String s : list) {
+            String[] parts = s.split(",");
+            if (parts[0].length() != 1) {
+                continue;
+            }
+            FruitTransaction.Operation mode;
+            switch (parts[0]) {
+                case "b" -> mode = FruitTransaction.Operation.BALANCE;
+                case "s" -> mode = FruitTransaction.Operation.SUPPLY;
+                case "p" -> mode = FruitTransaction.Operation.PURCHASE;
+                case "r" -> mode = FruitTransaction.Operation.RETURN;
+                default -> mode = null;
+            }
+            if (mode == null) {
+                continue;

```

Instead of silently continuing when an unknown operation code is encountered, consider logging a warning or throwing an exception to help identify data issues early.